dotnet-statsig/src/Statsig/Server/Evaluation/SpecStore.cs

```diff
@@ -24,104 +25,199 @@ internal IDList(string name)
 
     class SpecStore
     {
-        double _lastSyncTime;
-        RequestDispatcher _requestDispatcher;
-        Timer _syncTimer;
-        Timer _idListSyncTimer;
+        private double _lastSyncTime;
+        private readonly RequestDispatcher _requestDispatcher;
 
-        internal Dictionary<string, ConfigSpec> FeatureGates { get; set; }
-        internal Dictionary<string, ConfigSpec> DynamicConfigs { get; set; }
-        internal Dictionary<string, IDList> IDLists { get; set; }
+        private Task _syncIDListsTask;
+        private Task _syncValuesTask;
+        private readonly CancellationTokenSource _cts;
+
+        internal Dictionary<string, ConfigSpec> FeatureGates { get; private set; }
+        internal Dictionary<string, ConfigSpec> DynamicConfigs { get; private set; }
+        private readonly Dictionary<string, IDList> _idLists;
+        private readonly ReaderWriterLockSlim _idListLock;
```

Comment:
This would allow concurrent reading while writing is locked, correct? That's pretty cool if so

Reply:
Correct.  Multiple concurrent readers can share the lock.  Writers require exclusive access.